src/benchmark/file_io_read_micro_benchmark.cpp

```diff
@@ -16,16 +16,40 @@
 
 namespace hyrise {
 
-void read_data_using_read(const size_t from, const size_t to, int32_t fd, uint32_t* read_data_start) {
+
+void read_data_using_read(const size_t from, const size_t to, int32_t fd, uint32_t* read_data_start,
+                          std::atomic<bool>& threads_ready_to_be_executed) {
+  while (!threads_ready_to_be_executed) {}
   const auto uint32_t_size = ssize_t{sizeof(uint32_t)};
-  const auto bytes_to_read = static_cast<ssize_t>(uint32_t_size * (to - from));
-  lseek(fd, from * uint32_t_size, SEEK_SET);
-  Assert((read(fd, read_data_start + from, bytes_to_read) == bytes_to_read),
-         close_file_and_return_error_message(fd, "Read error: ", errno));
+  const auto total_bytes_to_read = static_cast<ssize_t>(uint32_t_size * (to - from));
```

Comment:
I don't know who started this, but ssize_t was never justified in the whole project.

Reply:
It is ssize_t because read / pread could -1 as an error code. At some point we have to cast them in order to compare it.